### DISCUSSION TITLE
how to delete rows from a many-to-many relation table ONLY

### DISCUSSION THREAD

**Discussion by saikarthikp9** [prisma/prisma#10459] [Q&A]
Hi, the following is my schema: 

```
model menuitem {
  id                 String               @id @default(uuid())
  name               String               @db.VarChar(100)
  price              Float
  menu_category      menu_category        @relation(fields: [category_id], references: [id], onDelete: Cascade, map: "menuitem_ibfk_1")
  menuitem_variation menuitem_variation[]
}

model menuitem_variation {
  id               String   @id @default(uuid())
  menuitem_id      String
  variation_id     String
  menuitem         menuitem @relation(fields: [menuitem_id], references: [id], onDelete: Cascade, map: "menuitem_variation_ibfk_1")
  variation        variation @relation(fields: [variation_id], references: [id], onDelete: Cascade, map: "menuitem_variation_ibfk_2")
}

model variation {
  id               String   @id @default(uuid())
  name             String   @db.VarChar(50)
  menuitem_variation menuitem_variation[]
}

```

I am trying to run the following query: 
```
db.menuitem_variation.deleteMany({
	where: {
		menuitem_id
	}
})
```
I get the following error:
```
PrismaClientKnownRequestError:
Invalid `prisma.menuitem_variation.deleteMany()` invocation:


  Foreign key constraint failed on the field: `menuitem_variation_ibfk_2 (index)`
    at cb (/prisma/generated/client/runtime/index.js:38735:17)
    at async Promise.all (index 0) {
  code: 'P2003',
  clientVersion: '3.2.1',
  meta: { field_name: 'menuitem_variation_ibfk_2 (index)' }
```

I also tried the following query based on the suggestion found here: [https://stackoverflow.com/questions/68670004/prisma-delete-many-to-many-relationship-with-composite-key](https://stackoverflow.com/questions/68670004/prisma-delete-many-to-many-relationship-with-composite-key), but I get the same error as above.
```
db.menuitem.update({
	data: {
		menuitem_variation: {
			deleteMany: {}
		}
	},
	where: {
		id: menuitem_id
	}
})
```

Prisma client version: 3.2.1
DB: postgresQL

The goal is to delete only the relationship row, not the menuitem or the variation row itself. Any help is greatly appreciated. Thanks!

**Comment by shibbirweb** [comment]
Here is documentation link to disconnect the relationship 

[Single Disconnect](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#disconnect-a-related-record)
```ts
const updatePost = await prisma.user.update({
  where: {
    id: 16,
  },
  data: {
    posts: {
      disconnect: [{ id: 12 }, { id: 19 }],
    },
  },
  select: {
    posts: true,
  },
})
```


[All disconnect](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#disconnect-all-related-records)

```ts
const updateUser = await prisma.user.update({
  where: {
    id: 16
  },
  data: {
    posts: {
      set: []
    }
  },
  include: {
    posts: true
  }
})
```
I prefer to disconnect all then connect to sync with the current relationship

```ts
const user = await prisma.user.update({
  where: {
    id: 1,
  },
  data: {
    email: 'vlad@prisma.io',
    posts: {
      set: [],
      connect: [{ id: 8 }, { id: 9 }, { id: 10 }],
    },
  },
  include: {
    posts: true, // Include all posts in the returned object
  },
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
